wsmaster/che-core-api-factory-azure-devops/src/main/java/org/eclipse/che/api/factory/server/azure/devops/AzureDevOpsUrl.java

```diff
@@ -145,7 +146,17 @@ public String rawFileLocation(String fileName) {
   }
 
   public String getRepositoryLocation() {
-    return getRepoPathJoiner().add("_git").add(repository).toString();
+    if (isHTTPSUrl) {
+      return getRepoPathJoiner().add("_git").add(repository).toString();
+    }
+    return "git@ssh."
+        + hostName.substring(8)
```

Comment:
I can see some magic numbers in the PR
Could you avoid that?

Reply:
done